src/v/pandaproxy/json/iobuf.h

```diff
@@ -57,4 +60,48 @@ class rjson_parse_impl<iobuf> {
     serialization_format _fmt;
 };
 
+template<>
+class rjson_serialize_impl<iobuf> {
+public:
+    explicit rjson_serialize_impl(serialization_format fmt)
+      : _fmt(fmt) {}
+
+    bool operator()(rapidjson::Writer<rapidjson::StringBuffer>& w, iobuf buf) {
+        switch (_fmt) {
+        case serialization_format::none:
+            [[fallthrough]];
+        case serialization_format::binary_v2:
+            return encode_base64(w, std::move(buf));
+        case serialization_format::unsupported:
+            [[fallthrough]];
+        default:
+            return false;
+        }
+    }
+
+    bool
+    encode_base64(rapidjson::Writer<rapidjson::StringBuffer>& w, iobuf buf) {
+        base64_state state{};
+        base64_stream_encode_init(&state, 0);
+        size_t out_len = (buf.size_bytes() * 4 + 2) / 3; // ceil
```

Comment:
did you find something more precise than what is in the documentation `it must be at least 4/3 the size of the in-buffer, but take some margin.` whatever `margin` means... though the README seems to be unambiguous when it comes to the requirement being at least 4/3.

given that, i think we should add assertions that we have not overflowed the bounds of `out`.

Reply:
You were right. It was overflowing in the test. The calculation is now right. But I added an assert for good measure.

Reply:
And a quick exit for empty buf.